models/maespa/R/met2model.MAESPA.R

```diff
@@ -0,0 +1,237 @@
+      RAD <-  ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") #W m-2
+      PAR <-   try(ncvar_get(nc,"surface_downwelling_photosynthetic_photon_flux_in_air")) #mol m-2 s-1
+      TAIR <-  ncvar_get(nc,"air_temperature") #K
+      `RH%` <- try(ncvar_get(nc,"relative_humidity")) #percentage
+      PPT <-  ncvar_get(nc,"precipitation_flux") #kg m-2 s-1
+      CA <- try(ncvar_get(nc,"mole_fraction_of_carbon_dioxide_in_air")) #mol/mol
+      PRESS <-  ncvar_get(nc,"air_pressure")# Pa
+      
+      # TSOIL <-  try(ncvar_get(nc,"soil_temperature")) #K
+      # VMFD <-  try(ncvar_get(nc,"vapour_pressure_mole_fraction_deficit")) 
+      # WIND <- try(ncvar_get(nc,"wind_speed")) #m/s   
+      # SW <-  try(ncvar_get(nc,"soil_moisture"))# kg m-2
+      # SWP <- ncvar_get(nc,"soil_water_potential)# MPa
+      # TDEW <- try(ncvar_get(nc, "dew_point_temperature")) #Celsius
+      # FBEAM <- ncvar_get try((nc,"fraction_of_surface_downwelling_photosynthetic_photon_flux_in_air") #frction of direct beam
+      # RH <- try(ncvar_get(nc,"relative_humidity"))# fraction
+      
+          if(!is.numeric(PAR)){
+            print("Photosynthetically Active Radiation(PAR)Calculated from RAD.If RAD absent, calculated from TAIR (DELTAT needed).")}else{
+              #4.6= unit conversion of W/m2 to mmol/m2/s
+              #0.4805 = based on the approximation that PAR is 0.45-0.50 of the total radiation
+              PAR <- PAR*4.6*0.4805##Convert W/m2 to mmol/m2/s 
+            }
+          # ºC   air temperature. If nonexistant. Error.
+          TAIR <- udunits2::ud.convert(TAIR,"kelvin","celsius") 
+          
+          ####ppm. atmospheric CO2 concentration. Constant from Enviiron namelist used instead
+          if(!is.numeric(CA)){print("Atmospheric CO2 concentration will be set to constant value set in ENVIRON namelist ")
+                              rm(CA)
+                              defaultCO2= 400 #400 is estimation of atmospheric CO2 in ppm)
+          }else{defaultCO2= 400} #400 is estimation of atmospheric CO2 in ppm))
+          
+          nc_close(nc)
+        } else {
+          print("Skipping to next year")
+          next
+        }
+        
+        ##build time variables (year, month, day of year)
+        skip = FALSE
+        nyr <- floor(length(sec)/86400/365*dt)
+        yr <- NULL
+        doy <- NULL
+        hr <- NULL
+        asec <- sec
+        for(y in year+1:nyr-1){
+          ytmp <- rep(y,365*86400/dt)
+          dtmp <- rep(1:365,each=86400/dt)
+          if(y %% 4 == 0){  ## is leap
+            ytmp <- rep(y,366*86400/dt)
+            dtmp <- rep(1:366,each=86400/dt)
+          }
+          if(is.null(yr)){
+            yr <- ytmp
+            doy <- dtmp
+            hr <- rep(NA,length(dtmp))
+          } else {
+            yr <- c(yr,ytmp)
+            doy <- c(doy,dtmp)
+            hr <- c(hr,rep(NA,length(dtmp)))
+          }
+          rng <- length(doy) - length(ytmp):1 + 1
```

Comment:
is it even possible for rng's elements to be less than 0 this way? `length(ytmp)` is always something positive, let's say 1460.  `length(doy)` is either equal to or greater than this. again `:` takes precedence so `length(ytmp):1` will be a sequence from `1460:1` (nothing zero yet). the biggest number in this sequence would either be equal to `length(doy)` or smaller. even when it's equal, you'll have a 0, but finally you're adding 1 which makes it impossible for rng's elements to be 0 or less than 0. I think what you want to evaluate is `(length(doy) - length(ytmp)):1 + 1` but this also can't give you anything less than 0 the way you set things up, right? as `length(doy) >=length(ytmp)` would always be the case, or am I missing something?